hdc/algo/accessors.py

```diff
@@ -540,22 +537,37 @@ def spi(
             )
 
         else:
-            groups = np.array(groups) if not isinstance(groups, np.ndarray) else groups
-            num_groups = np.unique(groups).size
 
-            if not groups.dtype.name == "int16":
-                warn("Casting groups to int16!")
-                groups = groups.astype("int16")
+            groups, keys = to_linspace(np.array(groups, dtype="str"))
```

Comment:
@Kirill888 the idea is that the user passes in a list / array / tuple / ... containing group labels which puts the `time` dimension into bins. The downstream function requires these groups to be in a linear space, so from `0` to `len(groups)-1` which was the expected input previously.

To take this burden off the user, this allows any labels and converts them to a `str` array, which is then converted to linear space. Maybe not most efficient, but for these applications totally fine - does that make sense for you?

For a practical example, we'll be using this to calculate SPIs for the full timeseries, grouping the time dimension into dekads per year. The user can then pass simply `xx.time.dekad.yidx` in without having to make sure the `groups` array is `0` indexed.